tests/api/v1/samples/upsertBulkAsGenerator.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/api/v1/samples/upsertBulkAsGenerator.js
+ */
+'use strict'; // eslint-disable-line strict
+const supertest = require('supertest');
+const api = supertest(require('../../../../index').app);
+const tu = require('../../../testUtils');
+const u = require('../collectors/utils');
+const Collector = tu.db.Collector;
+const expect = require('chai').expect;
+const Generator = tu.db.Generator;
+const GeneratorTemplate = tu.db.GeneratorTemplate;
+const sgUtils = require('../generators/utils');
+const gtUtil = sgUtils.gtUtil;
+const constants = require('../../../../api/v1/constants');
+
+describe('tests/api/v1/samples/upsertBulkAsGenerator.js >', () => {
+  let token;
+  let user;
+  const defaultCollector = u.getCollectorToCreate();
+  defaultCollector.version = '0.0.1';
+
+  let generatorWithCollector;
+  let generatorWithoutCollector;
+  let collector;
+  const generatorTemplate = gtUtil.getGeneratorTemplate();
+
+  before((done) => {
+    tu.createUserAndToken()
+      .then((createdUser) => {
+        user = createdUser.user;
+        token = createdUser.token;
+        return done();
+      })
+      .catch(done);
+  });
+
+  beforeEach((done) => {
+    sgUtils.createGeneratorAspects()
+      .then(() => GeneratorTemplate.create(generatorTemplate))
+      .then(() => Collector.create(u.getCollectorToCreate()))
+      .then((collectorCreated) => {
+        collector = collectorCreated;
+
+        const generator1 = sgUtils.getGenerator();
+        generator1.name += '-generator-1';
+        generator1.createdBy = user.id;
+        generator1.collectorId = collectorCreated.id;
+        generator1.possibleCollectors = [collector.name];
+
+        // Create template to the Generator
+        sgUtils.createSGtoSGTMapping(generatorTemplate, generator1);
+
+        const generator2 = sgUtils.getGenerator();
+        generator2.name += '-generator-2';
+        generator2.createdBy = user.id;
+        generator2.possibleCollectors = [collector.name];
+
+        return Generator.bulkCreate([generator1, generator2]);
+      })
+      .then((generators) => {
+        generatorWithCollector = generators[0];
+        generatorWithoutCollector = generators[1];
+        return collector.addPossibleGenerators(
+          [generatorWithCollector, generatorWithoutCollector]
+        );
+      })
+      .then(() => done())
+      .catch(done);
+  });
+
+  afterEach(u.forceDelete);
+  after(tu.forceDeleteUser);
+
+  describe('> with generator and collector running', () => {
```

Comment:
our convention is that every "describe" title ENDS WITH " >" -- not at the beginning of a describe title, and not in the "it" title